File: easy/testing.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    my ( $a, $b ) = split( / \| /, $line );
    my $r = 0;
    for ( my $i = 0 ; $i < length $a ; $i++ ) {
        $r++ if substr( $a, $i, 1 ) ne substr( $b, $i, 1 );
    }
    if ( $r == 0 ) {
        print "Done\n";
    }
    elsif ( $r < 3 ) {
        print "Low\n";
    }
    elsif ( $r < 5 ) {
        print "Medium\n";
    }
    elsif ( $r < 7 ) {
        print "High\n";
    }
    else {
        print "Critical\n";
    }
}
close(INFILE);

File: easy/testing.rb
m = %w(Done Low Medium High Critical)

def prio(a)
  case a
  when 0 then 0
  when 1..2 then 1
  when 3..4 then 2
  when 5..6 then 3
  else 4
  end
end

File.open(ARGV[0]).each_line do |line|
  s = line.chomp.split(' | ')
  r = 0
  (0...s[0].length).each do |i|
    r += 1 if s[0][i] != s[1][i]
  end
  puts m[prio(r)]
end
